airbyte-webapp/src/packages/cloud/services/auth/AuthService.tsx

```diff
@@ -64,6 +68,45 @@ type AuthContextApi = {
 
 export const AuthContext = React.createContext<AuthContextApi | null>(null);
 
+const getTempSignUpStorageKey = (currentUser: FbUser): string =>
+  `${currentUser.uid}/temp-signup-data`;
+const TempSignUpValuesProvider = {
+  get: async (
+    currentUser: FbUser
+  ): Promise<{
+    companyName: string;
+    name: string;
+    news: boolean;
+  }> => {
+    try {
+      const key = getTempSignUpStorageKey(currentUser);
+
+      const storedValue = localStorage.getItem(key);
+
+      if (storedValue) {
+        return JSON.parse(storedValue);
+      }
+    } catch (err) {
+      // passthrough and return default values
+    }
+
+    return {
+      companyName: "",
+      name: currentUser.email ?? "",
+      news: false,
+    };
+  },
+  save: async (
+    currentUser: FbUser,
+    v: { companyName: string; name: string; news: boolean }
+  ) => {
+    localStorage.setItem(
+      `${currentUser.uid}/temp-signup-data`,
```

Comment:
:art: We could also use `getTempSignUpStorageKey` here.

Reply:
Oh, exactly! - that's why that method was added.